### ISSUE TITLE
Setting sandbox is invalid: Expected field 'root'

### ISSUE THREAD

**Issue by rauanmayemir** [ocamllabs/vscode-ocaml-platform#245]
The sandbox is automatically set to 'global', so choosing esy will update `.vscode/settings.json` to:

```
  "ocaml.sandbox": {
    "root": "/Users/rauan/workspace/project",
    "kind": "esy"
  }
```

I don't want that because I want reproducible dev env for my teammates and thus commit `settings.json` to git. Setting `root` to my local path breaks that.

If I remove the `"root"` key, I get the error message:

```
Setting sandbox is invalid: Expected field 'root'
```

Since I already have a requirement to run `esy` as part of the env setup, I wish the plugin would recognize that and use it as the default. Or if it alternatively could use esy env from the cwd/root.

**Comment by rgrinberg** [comment]
I guess this plugin relies on `.vscode/settings.json` being private to your own environment rather than being shared. How would you expect the plugin to remember the choice you've made for which sandbox to use?

**Comment by rauanmayemir** [comment]
That's a good point.

I might have expected the plugin to give esy higher priority because having esy config would mean clear intention of using esy instead of global (admittedly, I'm biased). I.e do not remember the choice unless explicitly specified. Default is global _unless_ there's an esy config or opam config.

Another issue is using actual path instead of opting for something like `${workspaceFolder}` or `${env:OCAML_SANDBOX}` by default.

(I would argue that currently selected toolchain and the flattened path is a runtime information, not a setting.)

**Comment by rgrinberg** [comment]
> (I would argue that currently selected toolchain and the flattened path is a runtime information, not a setting.)

Still needs to be saved somewhere unless we'd like to use to make the selection every time when the default doesn't work.

I guess a better default is to use esy when `esy.json` is present. We could consider that.

**Comment by phated** [comment]
If `esy.json` detection is used, it should also be able to find (or at least be configured) to search for that file in different directories. For example: [grain](https://github.com/grain-lang/grain) is a monorepo where all the Reason/OCaml code lives in the `compiler` directory.

**Comment by tmattio** [comment]
This should have been fixed by the sandbox detection (#445), which will set the sandbox to Esy if there's an `esy.json` file in the workspace.
